engines/ultima/ultima8/ultima8.cpp

```diff
@@ -638,8 +638,8 @@ void Ultima8Engine::paint() {
 	}
 #endif
 
-	// Draw the mouse
-	_mouse->paint();
+	// Update the mouse
+	_mouse->update();
```

Comment:
If I follow the logic correctly here, `update()` is called on every game paint, so does that mean `CursorMan.replaceCursor` is called every frame?

I assume we can fix that by only doing something in update if something has changed.. or maybe this call isn't needed at all?

Reply:
Yes, this does call replace cursor every frame. The call to Mouse::flashCrossCursor() sets a timeout before the red cross cursor expires. Without updating here the red cross would remain on screen until another update occurs, which is likely to be mouse movement. There also might be a similar dependency for the combat cursor currently, although that could be solved by calling mouse update when combat is toggled.
 
I'm not sure if there's a gain to re-working this, but it might interesting to handle is a different manner. Possible answer could be to only update in the run loop on a set interval.

Reply:
No problem, I think the change is still worthwhile as it opens up the possibility of doing that sort of improvement in the future, but that would probably need more testing and consideration than my initial thought.